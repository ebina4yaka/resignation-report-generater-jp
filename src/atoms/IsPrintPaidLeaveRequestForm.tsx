import { ReactElement } from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'

type Props = {
  daysOfPaidLeaveRemaining: number
}

const useStyles = makeStyles(
  createStyles({
    root: {
      textAlign: 'left',
      color: 'silver',
    },
  })
)

export default function IsPrintPaidLeaveRequestForm(
  props: Props
): ReactElement {
  const { daysOfPaidLeaveRemaining } = props
  const classes = useStyles()
  if (daysOfPaidLeaveRemaining > 0) {
    return <div className={classes.root}>※別途有給取得の時季指定書あり</div>
  }
  return <></>
}
