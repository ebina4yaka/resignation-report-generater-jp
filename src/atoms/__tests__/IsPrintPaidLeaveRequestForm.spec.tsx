import React from 'react'
import renderer from 'react-test-renderer'
import IsPrintPaidLeaveRequestForm from '../IsPrintPaidLeaveRequestForm'

test('Snapshot test for PreviewDocxTitle daysOfPaidLeaveRemaining is 0', () => {
  const tree = renderer
    .create(<IsPrintPaidLeaveRequestForm daysOfPaidLeaveRemaining={0} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot test for PreviewDocxTitle daysOfPaidLeaveRemaining is 1', () => {
  const tree = renderer
    .create(<IsPrintPaidLeaveRequestForm daysOfPaidLeaveRemaining={1} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
