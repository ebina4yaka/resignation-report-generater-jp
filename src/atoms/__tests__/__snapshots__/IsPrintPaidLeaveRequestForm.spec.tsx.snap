// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for PreviewDocxTitle daysOfPaidLeaveRemaining is 0 1`] = `null`;

exports[`Snapshot test for PreviewDocxTitle daysOfPaidLeaveRemaining is 1 1`] = `
<div
  className="makeStyles-root-1"
>
  ※別途有給取得の時季指定書あり
</div>
`;
